src/cloudflare/internal/vectorize-api.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+/// <reference path="./vectorize.d.ts" />
+
+interface Fetcher {
+  fetch: typeof fetch;
+}
+
+enum Operation {
+  INDEX_GET,
+  VECTOR_QUERY,
+  VECTOR_INSERT,
+  VECTOR_UPSERT,
+  VECTOR_GET,
+  VECTOR_DELETE,
+}
+
+class VectorSearch {
+  public static async send(
+    fetcher: Fetcher,
+    operation: Operation,
+    endpoint: string,
+    init: RequestInit
+  ): Promise<Response> {
+    const res = await fetcher.fetch(
+      `http://vector-search/binding/${endpoint}`,
+      init
+    );
+    if (res.status !== 200) {
+      try {
+        const err = (await res.json()) as VectorizeError;
+        throw new Error(
+          `${Operation[operation]}_ERROR ${
+            typeof err?.code === "number" ? ` (code = ${err.code})` : ""
+          }: ${err.error}`,
+          {
+            cause: new Error(err.error),
+          }
+        );
+      } catch (e) {
+        throw new Error(
+          `${Operation[operation]}_ERROR: Status + ${res.status}`,
+          {
+            cause: new Error(`Status ${res.status}`),
+          }
+        );
+      }
+    }
+
+    return res;
+  }
+}
+
+class VectorizeIndexImpl implements VectorizeIndex {
+  public constructor(
+    private readonly fetcher: Fetcher,
+    private readonly indexId: string
+  ) {}
+
+  public async describe(): Promise<VectorizeIndexDetails> {
+    const res = await this._send(
+      Operation.INDEX_GET,
+      `indexes/${this.indexId}`,
+      {
+        method: "GET",
+      }
+    );
+
+    return (await res.json()) as VectorizeIndexDetails;
```

Comment:
If the `res.json()` fails with a JSON parsing error that will get directly propagated here to the caller of `describe()`. Is that what you want? Or should this be wrapped in an intermediate try/catch